### DISCUSSION TITLE
Extending Bespoke-fit on shorter polymers to longer molecules

### DISCUSSION THREAD

**Discussion by tlfobe** [openforcefield/openff-bespokefit#206] [Q&A] [answered]
Hello! I got around to using Bespoke-fit on non-biological heteropolymer systems I'm working on and I was wondering how transferable the generated bespoke force fields are to longer molecules made up of the same monomer? I'm running a bespoke fit on a 3-mer  (44 atoms per monomer, with a total of 153 atoms) but would like to extend the parameters optimized in the 3-mer to arbitrary N-mers. With how SMIRKS are defined, could I assign parameters to a longer N-mer using the output .xml file from the bespoke fit on the 3-mer or would I have to go in and make the SMIRKS string more general?

**Comment by Yoshanuikabundi** [comment, marked as answer]
Hi! At the end of the day, it'll depend on how your 3-mer is fragmented. For a linear polymer, I think you'll need at least three fragments along the backbone - 1 for each terminus, and a third for interior residues. If the fragmentation is right, the SMARTS should generalise to any molecule that has that fragment, which will include polymers of arbitrary length.

If the fragmentation is wrong, then the parameters themselves may not generalise to polymers - they might only match to the entire 3-mer backbone, for instance. That might be an indication that you need to optimise against a longer polymer, because interior residues depend on their neighbours for their electronic environment.

You can fine tune the size of your fragments by tweaking the fragmentation engine settings, but just be aware that if you're making the fragments smaller, the torsion drives will be missing some of the context they need to generate good parameters.

I'm excited to hear that you're using our tools on polymers! Let us know how it goes!
